clang/lib/CIR/Dialect/Transforms/LoweringPrepare.cpp

```diff
@@ -878,6 +886,14 @@ void LoweringPreparePass::lowerGlobalOp(GlobalOp op) {
            "custom initialization priority NYI");
     dynamicInitializers.push_back(f);
   }
+
+  // Collect global annotations
+  auto annotations = op.getAnnotations();
```

Comment:
Use of `auto` is really specific to a few scenarios, see https://llvm.github.io/clangir/GettingStarted/coding-guideline.html#use-of-auto, none of the uses in this snippet obeys that rule. Please make sure all uses of auto in this PR which do not obey the rules get changed. There are tons of those in LLVM lowering too.